logstash-core/src/main/java/org/logstash/FileLockFactory.java

```diff
@@ -95,4 +99,13 @@ public FileLock obtainLock(String lockDir, String lockName) throws IOException {
             throw new LockException("Lock held by this virtual machine: " + realLockPath);
         }
     }
+
+    public void releaseLock(FileLock lock) throws IOException {
```

Comment:
no that should not happen but we have no control on how you use this class really.